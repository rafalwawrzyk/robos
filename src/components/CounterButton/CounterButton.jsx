import React,{Component} from 'react'

 class CounterButton extends Component{
     constructor(props){
         super(props);
         this.state = {
             count:0
         }
     }
    shouldComponentUpdate(nextProps,nextState){
        if(nextState.count !== this.state.count){
            return true;
        }
        return false
    }

    updateCount = () => {
        const newState = {
            ...this.state,
            count:this.state.count + 1
        }
        this.setState(newState)
    }
    
    render(){
        return (
            <div>
              <button 
              color={this.props.color} 
              onClick={this.updateCount}
              id='counter'>
                Count:{this.state.count}
              </button>
            </div>
          )
    }

}


export default CounterButton;