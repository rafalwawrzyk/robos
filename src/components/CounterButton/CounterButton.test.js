import { shallow } from 'enzyme';
import React, { Component } from 'react';
import CounterButton from './CounterButton';

it('Check the counter component', () => {
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});


it('crectly increments the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count:1})
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count:2})
    expect(wrapper.instance().props.color).toEqual('red')
})




// describe('<CounterButton />', () => {
// 	it('should render button', () => {
// 		const wrapper = shallow(<CounterButton />);
// 		expect(wrapper.find('div').find('button'));
// 	});
// 	it('should click event', () => {
//         const state = 0;
// 		const button = shallow(
//         <CounterButton></CounterButton>);
//         expect(state).toEqual(0);
// 		button.find('button').simulate('click');
		
// 	});
// });

// describe('<MyComponent />', () => {
//     it('should render three <Foo /> components', () => {
//       const wrapper = shallow(<MyComponent />);
//       expect(wrapper.find(Foo)).to.have.length(3);
//     });

//     it('should render an `.icon-star`', () => {
//       const wrapper = shallow(<MyComponent />);
//       expect(wrapper.find('.icon-star')).to.have.length(1);
//     });

//     it('should render children when passed in', () => {
//       const wrapper = shallow((
//         <MyComponent>
//           <div className="unique" />
//         </MyComponent>
//       ));
//       expect(wrapper.contains(<div className="unique" />)).to.equal(true);
//     });

//     it('simulates click events', () => {
//       const onButtonClick = sinon.spy();
//       const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
//       wrapper.find('button').simulate('click');
//       expect(onButtonClick.calledOnce).to.equal(true);
//     });
//   });
