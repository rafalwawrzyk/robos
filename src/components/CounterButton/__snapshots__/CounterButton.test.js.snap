// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the counter component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton
    color="red"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        color="red"
        id="counter"
        onClick={[Function]}
>
        Count:
        0
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count:",
          0,
        ],
        "color": "red",
        "id": "counter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count:",
        0,
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          color="red"
          id="counter"
          onClick={[Function]}
>
          Count:
          0
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Count:",
            0,
          ],
          "color": "red",
          "id": "counter",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Count:",
          0,
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
